src/internal/core.ts

```diff
@@ -440,19 +447,10 @@ export const async = <R, E, A>(
 ): Effect.Effect<R, E, A> =>
   suspend(() => {
     let cancelerRef: Effect.Effect<R, never, void> | void = undefined
-    let controllerRef: AbortController | void = undefined
+    let controllerRef: AbortController | undefined = undefined
     const effect = new EffectPrimitive(OpCodes.OP_ASYNC) as any
-    if (register.length !== 1) {
```

Comment:
I prefer the previous version, as it it was self-contained.

If `withSignalAndCallback` was used in 3 or more places, then I think the change could be worth it.